art_tree.go

```diff
@@ -17,40 +20,90 @@ func NewArtTree() *ArtTree {
 	return &ArtTree{root: nil, size: 0}
 }
 
+type Result struct {
+	Key   string
+	Value interface{}
+}
```

Comment:
Sorry for the slow progress on that. I got pulled into other work and have only just gotten down far enough to context switch back. Prefix search was going to be my NEXT PR, as that is what I really need, so I will take a look at this one now.